### PR TITLE
Repo Setup

### PR DESCRIPTION
**Pull request by zombie-einstein**
- Add poetry files
- Add pre-commit settings
- Add test
- Refactor module for linting and correct structure

### CODE REVIEW

File: jax_tqdm/pbar.py [hunk 1]
```diff
@@ -1,77 +1,91 @@
...
         tqdm_bars[0].set_description(message, refresh=False)
 
     def _update_tqdm(arg, transform):
         tqdm_bars[0].update(arg)
-    
+
     def _update_progress_bar(iter_num):
         "Updates tqdm progress bar of a JAX scan or loop"
-        _ = lax.cond(
+        _ = jax.jax.lax.cond(
```
Comment (jeremiecoullon) [inline jax_tqdm/pbar.py:43]: one too many `jax`'s :)

File: jax_tqdm/pbar.py [hunk 2]
```diff
@@ -1,77 +1,91 @@
...
             operand=None,
         )
-        
 
     def _progress_bar_scan(func):
-        """Decorator that adds a progress bar to `body_fun` used in `lax.scan`.
-        Note that `body_fun` must either be looping over `np.arange(num_samples)`,
-        or be looping over a tuple who's first element is `np.arange(num_samples)`
+        """Decorator that adds a progress bar to `body_fun` used in `jax.lax.scan`.
+        Note that `body_fun` must either be looping over `np.arange(n)`,
```
Comment (jeremiecoullon) [inline jax_tqdm/pbar.py:79]: should probably be `jax.numpy.arange` rather than numpy.arange? (same for the next line)

**Review by jeremiecoullon — CHANGES_REQUESTED** [review]
just a small bug fix, then all good!

File: jax_tqdm/pbar.py (hunk 1 above)
Comment (zombie-einstein) [reply jax_tqdm/pbar.py:43]: Yes good spot!

File: jax_tqdm/pbar.py (hunk 2 above)
Comment (zombie-einstein) [reply jax_tqdm/pbar.py:79]: Yup will change

**Review by jeremiecoullon — APPROVED** [review]
LGTM.

(Also, in general it might be nice to use squash and merge rather than merge)

**Comment by zombie-einstein** [comment]
> LGTM.
> 
> (Also, in general it might be nice to use squash and merge rather than merge)

Cool yep agreed

**Pull request merged**